Ports/llvm/patches/llvm.patch

```diff
@@ -0,0 +1,214 @@
+ 
+ /**
+ * Converts a std::wstring to a UTF-8 encoded std::string.
+ * \return true on success.
+ */
+-bool convertWideToUTF8(const std::wstring &Source, std::string &Result);
++bool convertWideToUTF8(const std::string &Source, std::string &Result);
+ 
+ 
+ /**
+diff -ruN llvm-orig/llvm-project-llvmorg-12.0.0/llvm/include/llvm/Support/SwapByteOrder.h llvm-project-llvmorg-12.0.0/llvm/include/llvm/Support/SwapByteOrder.h
+--- llvm-orig/llvm-project-llvmorg-12.0.0/llvm/include/llvm/Support/SwapByteOrder.h	2021-04-06 19:38:18.000000000 +0300
++++ llvm-project-llvmorg-12.0.0/llvm/include/llvm/Support/SwapByteOrder.h	2021-06-09 16:00:20.111549941 +0300
+@@ -22,7 +22,7 @@
+ #endif
+ 
+ #if defined(__linux__) || defined(__GNU__) || defined(__HAIKU__) ||            \
+-    defined(__Fuchsia__) || defined(__EMSCRIPTEN__)
++    defined(__Fuchsia__) || defined(__EMSCRIPTEN__) || defined(__serenity__)
+ #include <endian.h>
+ #elif defined(_AIX)
+ #include <sys/machine.h>
+diff -ruN llvm-orig/llvm-project-llvmorg-12.0.0/llvm/lib/Support/ConvertUTFWrapper.cpp llvm-project-llvmorg-12.0.0/llvm/lib/Support/ConvertUTFWrapper.cpp
+--- llvm-orig/llvm-project-llvmorg-12.0.0/llvm/lib/Support/ConvertUTFWrapper.cpp	2021-04-06 19:38:18.000000000 +0300
++++ llvm-project-llvmorg-12.0.0/llvm/lib/Support/ConvertUTFWrapper.cpp	2021-06-09 16:15:53.330400711 +0300
+@@ -196,15 +196,15 @@
+     Result.clear();
+     return false;
+   }
+-  Result.resize(reinterpret_cast<wchar_t *>(ResultPtr) - &Result[0]);
++  Result.resize(reinterpret_cast<char *>(ResultPtr) - &Result[0]);
+   return true;
+ }
+ 
+-bool ConvertUTF8toWide(llvm::StringRef Source, std::wstring &Result) {
++bool ConvertUTF8toWide(llvm::StringRef Source, std::string &Result) {
+   return ConvertUTF8toWideInternal(Source, Result);
+ }
+ 
+-bool ConvertUTF8toWide(const char *Source, std::wstring &Result) {
++bool ConvertUTF8toWide(const char *Source, std::string &Result) {
+   if (!Source) {
+     Result.clear();
+     return true;
+@@ -212,7 +212,7 @@
+   return ConvertUTF8toWide(llvm::StringRef(Source), Result);
+ }
+ 
+-bool convertWideToUTF8(const std::wstring &Source, std::string &Result) {
++bool convertWideToUTF8(const std::string &Source, std::string &Result) {
+   if (sizeof(wchar_t) == 1) {
+     const UTF8 *Start = reinterpret_cast<const UTF8 *>(Source.data());
+     const UTF8 *End =
+diff -ruN llvm-orig/llvm-project-llvmorg-12.0.0/llvm/lib/Support/CrashRecoveryContext.cpp llvm-project-llvmorg-12.0.0/llvm/lib/Support/CrashRecoveryContext.cpp
+--- llvm-orig/llvm-project-llvmorg-12.0.0/llvm/lib/Support/CrashRecoveryContext.cpp	2021-04-06 19:38:18.000000000 +0300
++++ llvm-project-llvmorg-12.0.0/llvm/lib/Support/CrashRecoveryContext.cpp	2021-06-09 16:16:18.978582822 +0300
+@@ -382,7 +382,7 @@
+ 
+   // Don't consider a broken pipe as a crash (see clang/lib/Driver/Driver.cpp)
+   if (Signal == SIGPIPE)
+-    RetCode = EX_IOERR;
++    RetCode = EX_IOER;
```

Comment:
Why is ours called `EX_IOER` anyway? :thinking:

musl and glibc both have `EX_IOERR`, this seems like a compatibility issue (and is, as shown here...)